optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/DefaultSolverFactory.java

```diff
@@ -65,24 +65,27 @@
     private static final long DEFAULT_RANDOM_SEED = 0L;
 
     private final SolverConfig solverConfig;
+    private volatile InnerScoreDirectorFactory<Solution_, ?> scoreDirectorFactory;
 
     public DefaultSolverFactory(SolverConfig solverConfig) {
         this.solverConfig = Objects.requireNonNull(solverConfig, "The solverConfig (" + solverConfig + ") cannot be null.");
     }
 
     public InnerScoreDirectorFactory<Solution_, ?> getScoreDirectorFactory() {
```

Comment:
I considered whether or not to make this lazy and I decided not to do the work in the constructor. It seems the user should only be paying the penalty if they actually ask for the solver. That said, I am open to reconsidering, if you have a better argument.